pkg/provider/aws/resources/eks.go

```diff
@@ -283,6 +286,37 @@ func createAddOns(clusterName string, provenance []core.AnnotationKey) []*kubern
 				},
 			},
 		},
+		&HelmChartAlbController{
+			HelmChart: kubernetes.HelmChart{
+
+				Name:             clusterName + `-alb-c`,
+				Chart:            `aws-load-balancer-controller`,
+				ConstructRefs:    provenance,
+				Repo:             `https://aws.github.io/eks-charts`,
+				Version:          `1.4.7`,
+				ClustersProvider: nil,       // TODO?
+				Namespace:        "default", // kube-system if on fargate, default otherwise
+			},
+			ClusterName: clusterName,
+			Region:      NewRegion(),
+			Vpc:         vpc,
+			Role: NewIamRole(clusterName, `alb-controller-role`, provenance, &PolicyDocument{
+				Version: VERSION,
+				Statement: []StatementEntry{
+					{
+						Effect:    `Allow`,
+						Action:    []string{`sts:AssumeRoleWithWebIdentity`},
+						Resource:  nil,
+						Principal: nil,
+						Condition: &Condition{
+							StringEquals: map[core.IaCValue]string{
+								nil: fmt.Sprintf(`system:serviceaccount:${namespace}:${name}`), // TODO fill out the key, and namenamespace, and name
+							},
+						},
```

Comment:
id maybe make the code which is in execution_unit.go reusable for generating the service account roles because we do that already and its a bit annoying to generate the weird assume role policy